flask_oauthres/oauth2.py

```diff
@@ -0,0 +1,155 @@
+
+        if hasattr(request, 'oauth') and request.oauth:
+            resp = request.oauth
+            token = request.oauth.get('access_token')
+        else:
+            if 'Authorization' in request.headers:
+                token = request.headers.get('Authorization')[7:]
+            else:
+                token = request.values.get('token', None)
+            resp = request.oauth if hasattr(request, 'oauth') and request.oauth else service.check_token(token)
+
+        if not resp:
+            return None
+
+        request.oauth = resp
+        if self.service.resource_id:
+            resources = resp.get('resources', [])
+            if self.service.resource_id not in resources:
+                log.debug("Resource=%s is not allowed for token=%s" % (self.service.resource_id, token))
+                return None
+
+        if scopes:
+            token_scopes = resp.get('scopes', [])
+            for scope in scopes:
+                if scope not in token_scopes:
+                    log.debug("Missing scope=%s" % scope)
+                    return None
+
+        if roles:
+            token_roles = resp.get('roles', [])
+            for role in roles:
+                if role not in token_roles:
+                    log.debug("Missing role=%s" % role)
+                    return None
+
+        return resp
+
+    def has_access(self):
+        """Protect resource without checking roles nor scopes."""
+        def wrapper(f):
+            @wraps(f)
+            def decorated(*args, **kwargs):
+                resp = self.verify_request(request)
+                if not resp:
+                    return abort(401)
+                return f(*args, **kwargs)
+            return decorated
+        return wrapper
+
+    def has_scope(self, *scopes):
+        """Protect resource with specified scopes."""
+        def wrapper(f):
+            @wraps(f)
+            def decorated(*args, **kwargs):
+                resp = self.verify_request(request, scopes=scopes)
+                if not resp:
+                    return abort(401)
+                return f(*args, **kwargs)
+            return decorated
+        return wrapper
+
+    def has_role(self, *roles):
+        """Protect resource with specified user roles."""
+        def wrapper(f):
+            @wraps(f)
+            def decorated(*args, **kwargs):
+                resp = self.verify_request(request, roles=roles)
+                if not resp:
+                    return abort(401)
+                return f(*args, **kwargs)
+            return decorated
+        return wrapper
+
+
+class OAuth2RemoteTokenService:
+
+    def __init__(self, resource_id, client_id, client_secret, check_token_endpoint_url):
+        self.resource_id = resource_id
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.check_token_endpoint_url = check_token_endpoint_url
+
+    def check_token(self, token):
+
+        if not token:
+            return None
```

Comment:
A już na pewno nie powinieneś się spodziewać None w argumencie check_token chyba, że to kwestia zewnętrznego api.